include/verilated_types.h

```diff
@@ -1003,12 +1004,159 @@ std::string VL_TO_STRING(const VlUnpacked<T_Value, T_Depth>& obj) {
     return obj.to_string();
 }
 
+class VlClass;  // See below
+
+//===================================================================
+// Class providing delayed deletion of garbage objects. Objects get deleted only when 'deleteAll()'
+// is called, or the deleter itself is destroyed.
+
+class VlDeleter final {
+    // MEMBERS
+    std::vector<VlClass*> m_newGarbage VL_GUARDED_BY(m_mutex);  // Queue of new objects to delete
+    std::vector<VlClass*> m_toDelete VL_GUARDED_BY(m_mutex);  // Queue of objects currently being
```

Comment:
Don't think m_toDelete needs to be guarded?

Reply:
Wilson suggested it does :) I'm not sure how smart the thread safety checks are. Can I run a test with those easily?

Reply:
The normal clang compile checks these.  std::swap isn't thread safe and you currently get the lock everywhere m_toDelete is used, which is why I think it needed guarding.

Reply:
Right, but `m_toDelete` is only modified by one thread. I think all this `guarded_by` attribute does is ensure we get the lock before we modify the guarded variable? It should work without `guarded_by`, but it's not really a problem to leave it.

Reply:
What is wrong with this?
```
     while (true) {
         {
             LockGuard lock{m_mutex};
￼            std::swap(m_newGarbage, m_toDelete);
         }
         if (m_toDelete.empty()) break;
￼        for (VlClass* const objp : m_toDelete) delete objp;
￼        m_toDelete.clear();
￼    }
``

Reply:
Nothing. Changed it to your solution.